apps/aehyperchains/src/aehc_parent_mng.erl

```diff
@@ -0,0 +1,333 @@
+%%%     b) The child chain is reverted and the new election process is started from the last synced track height;
+%%% - To be able to run the Hyperchains the system must satisfy the connector's acceptance criteria;
+%%%     a) For default mode: get_top_block/0, get_block_by_hash/1l
+%%%     b) For delegate mode: default mode + send_tx/1;
+%%% - Each interested developer can supply his own connector's implementation for the particular parent chain;
+%%% (TODO: To supply link to the official connector's development guide);
+%%% @end
+%%%-------------------------------------------------------------------
+-module(aehc_parent_mng).
+
+-behaviour(gen_server).
+
+%% API.
+-export([start_link/0]).
+-export([accept_connector/1]).
+
+-export([publish_block/2]).
+
+%% gen_server.
+-export([init/1]).
+-export([handle_call/3]).
+-export([handle_cast/2]).
+-export([handle_info/2]).
+-export([terminate/2]).
+
+-include_lib("aeutils/include/aeu_stacktrace.hrl").
+
+%% The track record represents the current synchronized view of a particular parent chain within manager;
+-record(track, {
+        %% The current syncronized top block hash from the Db;
+        current_hash::binary(),
+        %% The previous syncronized top block hash from the Db;
+        previous_hash::binary(),
+        %% The current syncronized top block height from the Db;
+        current_height::non_neg_integer(),
+        %% The genesis hash from the config;
+        genesis_hash :: binary(),
+        %% Responsible connector module;
+        connector :: aehc_connector:connector(),
+        %% Passed args;
+        args :: map(),
+        %% Textual description;
+        note :: binary()
+    }).
+
+-type track() :: #track{}.
+%% API.
+
+-spec start_link() ->
+    {ok, pid()} | ingnore | {error, term()}.
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+%% This event issued each time by connector when the new block is generated;
+-spec publish_block(aehc_connector:connector(), aehc_parent_block:parent_block()) -> ok.
+publish_block(Connector, Block) ->
+    erlang:send(?MODULE, {publish_block, Connector, Block}).
+
+%%%===================================================================
+%%%  gen_server behaviour
+%%%===================================================================
+
+%% TODO: Connector check API;
+-record(state, { master :: aehc_connector:connector(), tracks :: list() }).
+
+init([]) ->
+    %% TODO: To extract the current block from Db;
+    process_flag(trap_exit, true),
+    %% Read configuration;;
+    Tracks = [conf_track(C) || C <- tracks_config()],
+    %% Db initialization
+    InitTracks = [init_db_track(Track) || Track <- Tracks],
+    %% Execute acceptance procedure;
+    [accept_connector(T) || T <- InitTracks],
+    %% Run conenctor's instances;
+    [start_connector(T) || T <- InitTracks],
+    %% Forks synchronization by fetched blocks;
+    SynchedTracks = [sync_track(T) || T <- InitTracks],
```

Comment:
This is blocking... This needs to be done in separate processes

Reply:
Agree, manager should operate independent processes, sure